src/feature.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  EvaluationApi,
+  EvaluationBatchRequest,
+  EvaluationBatchRequestFlagTagsOperatorEnum,
+  EvaluationEntity,
+  createConfiguration,
+  ServerConfiguration,
+} from "flagr-client";
+
+export interface FlagCallbacks {
+  [key: string]: (...args: any) => Promise<any>;
+}
```

Comment:
**suggestion:** we can make this stricter with generics
```suggestion
export interface FlagCallbacks<T> {
  [key: string]: () => Promise<T>;
}
```
This does two things- it ensures the functions always have _some_ return type, and restricts them both(/all?) to the same return type.

```ts
// error
{
  on: () => 'foo',
  otherwise: () => false
}
// good
{
  on: () => 'foo',
  otherwise: () => 'bar'
}
```

Also, I assume the args aren't needed?

Reply:
Another one- you can do this so you don't need `async` on normal functions
```tsx
export interface FlagCallbacks<T> {
  [key: string]: () => (T | Promise<T>);
}